src/main/java/core/basesyntax/Service/DataConverterImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.service;
+
+import core.basesyntax.DataConverter;
+import core.basesyntax.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String CSV_HEADER = "type,fruit,quantity";
+    private static final int MAX_NUMBER_OF_PARTS = 3;
+    private static final int INDEX_OF_TYPE_CODE = 0;
+    private static final int INDEX_OF_FRUIT = 1;
+    private static final int INDEX_OF_QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.trim().isEmpty() || line.equals(CSV_HEADER)) {
+                continue;
+            }
+            String[] parts = line.split(",");

```

Comment:
i would suggest extracting logic of parsing one line into a separate method and call it here